File: app/src/main/java/eugeen3/keepinfit/dao/FoodItemDAO.java
package eugeen3.keepinfit.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import eugeen3.keepinfit.entities.FoodItem;

@Dao
public interface FoodItemDAO {

    @Insert
    void insert(FoodItem foodItem);

    @Update
    void update(FoodItem foodItem);

    @Delete
    void delete(FoodItem foodItem);

    @Query("DELETE FROM foodItems_table")
    void deleteAll();

    @Query("SELECT * FROM foodItems_table ORDER BY name ASC")
    LiveData<List<FoodItem>> getAllFoodItems();
}

File: app/src/main/java/eugeen3/keepinfit/dao/FoodItemDAO.kt
package eugeen3.keepinfit.dao

import eugeen3.keepinfit.entities.FoodItem
import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface FoodItemDAO {
    @Insert
    fun insert(foodItem: FoodItem)

    @Update
    fun update(foodItem: FoodItem)

    @Delete
    fun delete(foodItem: FoodItem)

    @Query("DELETE FROM foodItems_table")
    fun deleteAll()

    @get:Query("SELECT * FROM foodItems_table ORDER BY name ASC")
    val allFoodItems: LiveData<List<FoodItem>>
}